server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/DelegateSetContract.scala

```diff
@@ -0,0 +1,515 @@
+    val request =
+      s"""{
+         |	"using": ["urn:ietf:params:jmap:core", "urn:apache:james:params:jmap:delegation"],
+         |	"methodCalls": [
+         |		[
+         |			"Delegate/set", {
+         |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |				"create": {
+         |					"4f29": {
+         |						"username": "nonexistuser@domain.tld"
+         |					}
+         |				}
+         |			}, "0"
+         |		]
+         |	]
+         |}""".stripMargin
+
+    val response =  `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .isEqualTo(
+        s"""{
+           |	"sessionState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
+           |	"methodResponses": [
+           |		[
+           |			"Delegate/set",
+           |			{
+           |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |				"newState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
+           |				"notCreated": {
+           |					"4f29": {
+           |						"type": "invalidArguments",
+           |						"description": "User nonexistuser@domain.tld does not exist"
+           |					}
+           |				}
+           |			},
+           |			"0"
+           |		]
+           |	]
+           |}""".stripMargin)
+  }
+
+  @Test
+  def delegateSetShouldBeIdempotent(server: GuiceJamesServer): Unit = {
+    val request =
+      s"""{
+         |	"using": ["urn:ietf:params:jmap:core", "urn:apache:james:params:jmap:delegation"],
+         |	"methodCalls": [
+         |		[
+         |			"Delegate/set", {
+         |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |				"create": {
+         |					"4f29": {
+         |						"username": "andre@domain.tld"
+         |					},
+         |					"4f30": {
+         |						"username": "andre@domain.tld"
+         |					}
+         |				}
+         |			}, "0"
+         |		]
+         |	]
+         |}""".stripMargin
+
+    val response =  `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+    val delegationId = DelegationId.from(BOB, ANDRE).serialize
+
+    assertThatJson(response)
+      .isEqualTo(
+        s"""{
+           |	"sessionState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
+           |	"methodResponses": [
+           |		["Delegate/set", {
+           |			"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |			"newState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
+           |			"created": {
+           |				"4f29": {
+           |					"id": "$delegationId"
```

Comment:
I don't think the return same ids make sense. 
I bet it is not clear in spec. 
I tend to return 1 id, 1 clientId, and precondition check in client request if need. 
Just my view

Reply:
I dont understand your point. Why do u want to return different `delegationId` while it is the same delegation?

Reply:
My mean is we can check request to do not accept same delegations per request.

Reply:
No need IMO, why do we need to avoid user creating the same delegation while we have idempotent?

Reply:
If we are idempotent then this result is expected (the two call says they prformed the entry creation thus reports the same id. Good to me.